### PR TITLE
chore: add CONTRIBUTING.md + create issue/PR templates

### PR DESCRIPTION
**Pull request by thelostone-mc**
### Description

- Adds a contributing.md guide to explain 
  - the overall process we follow for discussions/issues/PR
  - how to start contributing
- Adds Pull Request Template
- Adds Issue Template for
  - bug
  - task

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+name: "\U0001F41B Bug report"
+about: Create a report to help us improve
```
Comment (mds1) [inline .github/ISSUE_TEMPLATE/bug_report.md:3]: maybe change "report" to "bug report"? 

```suggestion
about: Create a bug report to help us improve
```

File: .github/ISSUE_TEMPLATE/task.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+name: "Tracking Task"
```
Comment (mds1) [inline .github/ISSUE_TEMPLATE/task.md:2]: Is "task" a catch-all for everything non-bug related? Seems like it, which makes sense, but just want to confirm

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+# Contributing to dGrants
+
+A big welcome and thank you for considering contributing to dGrants open source projects! It’s people like you that make it a reality for users in our community.
```
Comment (mds1) [inline CONTRIBUTING.md:3]: This is the only repo right now, so just changing from plural to singular 

```suggestion
A big welcome and thank you for considering contributing to the dGrants open source project! It’s people like you that make it a reality for users in our community.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    * [Issues](#issues)
+    * [Pull Requests](#pull-requests)
+      * [Git Rebase Workflow](#git-rebase-workflow)
+* [Getting Help](#getting-help)
+
+## Code of Conduct
+
+Contributions to Gitcoin are governed by the [Contributor Covenant version 1.4](https://www.contributor-covenant.org/version/1/4/code-of-conduct.html).
+All contributors and participants agree to abide by its terms. To report
+violations, shoot out an email to founders@gitcoin.co
```
Comment (mds1) [inline CONTRIBUTING.md:21]: "shoot out an email to founders@gitcoin.co" feels like it's not the reporting process we want for decentralized grants, but I don't have a good alternative to propose 🤔

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+Open, diverse, and inclusive communities live and die on the basis of trust.
+Contributors can disagree with one another so long as they are done in good
+faith and everyone is working towards a common goal.
+
+## Getting Started
+
+Contributions are made to this repo via Discussions, Issues and Pull Requests (PRs). A few general guidelines that cover the trio:
+
+- Search for existing Issues and PRs before creating your own.
```
Comment (mds1) [inline CONTRIBUTING.md:35]:
```suggestion
- Search for existing Discussions, Issues, and PRs before creating your own.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+- We work hard to makes sure issues are handled in a timely manner but, depending on the impact, it could take a while to investigate the root cause. A friendly ping in the comment thread to the submitter or a contributor can help draw attention if your issue is blocking.
+
+### Discussions
+
+Discussion should be used to discuss potential changes before a PR is created,  or discussing the implmentation on a current feature. Once the discussion has ended with participation from the community and maintainers, if changes need to made to the codebase, follow it up by creating an issue.
```
Comment (mds1) [inline CONTRIBUTING.md:40]: IIRC we agreed the flow would almost always be discussion > issue > PR, so just clarifying that here. Also fixing some typos

```suggestion
Discussion should be used to discuss potential changes before an issue or PR is created, or discussing the implementation on a current feature. Once the discussion has ended with participation from the community and maintainers, if changes need to made to the codebase, follow it up by creating an issue.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+### Issues
+
+Issues should be used to report problems with the dApp, request a new feature, update an existing feature (which has a discussion on the discussion board). When you create a new Issue, a template will be loaded that will guide you through collecting and providing the information we need to investigate.
```
Comment (mds1) [inline CONTRIBUTING.md:44]: Similar to above 
```suggestion
Issues should be used to report problems with the dApp, request a new feature, or update an existing feature (and features should have a discussion on the discussion board before creating an issue). When you create a new Issue, a template will be loaded that will guide you through collecting and providing the information we need to investigate.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+If you find an Issue that addresses the problem you're having, please add your own reproduction information to the existing issue rather than creating a new one. Adding a [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) can also help be indicating to our maintainers that a particular problem is affecting more than just the reporter.
+
+When creating a new issue, please do your best to be as detailed and specific as possible.
+
+### Pull Requests
+
+PRs to our repo are always welcome and can be a quick way to get your fix or improvement slated for the next release. In general, PRs should:
+
+- Should have a correspoding issue created.
+- Only fix/add the functionality described on the issue to ensure reviewers are cleaner
```
Comment (mds1) [inline CONTRIBUTING.md:55]:
```suggestion
- Only fix/add the functionality described on the issue to ensure reviews are clean and well-scoped
```

**Review by mds1 — DISMISSED** [review]
This looks awesome! Just left a few small comments

File: .github/ISSUE_TEMPLATE/task.md (hunk 2 above)
Comment (thelostone-mc) [reply .github/ISSUE_TEMPLATE/task.md:2]: Yup that's right ! Having a to create one for feature / task / etc seemed overkill at this point in time

**Comment by thelostone-mc** [comment]
Ah thanks for the review @mds1 ! Updated the PR

File: CONTRIBUTING.md (hunk 4 above)
Comment (phutchins) [reply CONTRIBUTING.md:21]: Yes, for the moment we can set it to my email. We could use a decentralized email account soon. I'm thinking https://ethmail.cc/ using the multisig potentially although I'm not sure how we would log in with the multisig.

**Pull request merged**